src/main/kotlin/se/uulm/snowballr/backend/service/PaperService.kt

```diff
@@ -38,9 +53,51 @@ class PaperService(
         val paperId = parseUUID(request.id, EntityType.PAPER)
         val paper = repo.getPaperById(paperId)
 
-        val authors = authorOfPapersRepo.getAuthorsOfPaperById(paperId).map { it.toGrpcAuthor() }
+        val authors = authorOfPapersRepo.getAuthorsOfPaperById(paperId)
         val backwardReferencedIds = citationRepo.getBackwardsReferencedPaperIdsOfPaperById(paperId)
-            .map { it.toString() }
-        return paper.toGrpcPaper(authors, backwardReferencedIds)
+        return paper.toGrpcPaper(
+            authors.map { it.toGrpcAuthor() },
+            backwardReferencedIds.map { it.toString() },
+        )
+    }
+
+    override suspend fun getBackwardReferencedPapers(request: Base.Id): GrpcPaper.List =
+        getReferencePapers(request, citationRepo::getBackwardsReferencedPaperIdsOfPaperById)
+
+    override suspend fun getForwardReferencedPapers(request: Base.Id): GrpcPaper.List =
+        getReferencePapers(request, citationRepo::getForwardReferencedPaperIdsOfPaperById)
+
+    /**
+     * Retrieves a list of reference papers based on the provided paper ID and a specified function for fetching references.
+     * This method ensures the validity of the paper ID and retrieves the associated metadata for each reference paper,
+     * including authors and backward references.
+     *
+     * @param request The request containing the ID of the paper for which references are to be retrieved.
+     * @param function A suspendable function that takes a UUID (representing a paper ID) and returns a list of UUIDs for the references.
+     * @return A list of gRPC-compatible paper objects containing reference information.
+     * @throws SnowballRException.NotFoundException If the paper specified in the request does not exist.
+     */
```

Comment:
Maximum line length.
```suggestion
    /**
     * Retrieves a list of reference papers based on the provided paper ID and a specified function for fetching
     * references. This method ensures the validity of the paper ID and retrieves the associated metadata for each
     * reference paper, including authors and backward references.
     *
     * @param request The request containing the ID of the paper for which references are to be retrieved.
     * @param function A function that takes a paper ID and returns a list of UUIDs of the references.
     * @return A list of gRPC-compatible paper objects containing reference information.
     * @throws SnowballRException.NotFoundException If the paper specified in the request does not exist.
     */
```